### PR TITLE
Feat: simplify questionnaire with references for public-enemy

### PR DESCRIPTION
**Pull request by laurentC35**
- Move dereference implementation directy to QuestionnaireService
- Create new method : `getQuestionnaireByIDWithReferences` to get directyl the questionnaire with its references
- enhancement of `/api/questionnaire/{id}` to allow new parameters 'references' to use the new questionnaireService method

Enhancement for Public-enemy, you can see this PR: https://github.com/InseeFr/Public-Enemy-Back-Office/pull/79

### CODE REVIEW

File: src/main/java/fr/insee/pogues/persistence/service/QuestionnairesServiceImpl.java [hunk 1]
```diff
@@ -109,4 +137,33 @@ public void updateJsonLunatic(String id, JSONObject dataLunatic) throws Exceptio
 	        throw new PoguesException(404, "Not found", e.getMessage());
 	    }
 	}
+
+	public Questionnaire deReference(JSONObject jsonQuestionnaire) throws Exception {
+
+		Questionnaire questionnaire = PoguesDeserializer.questionnaireToJavaObject(jsonQuestionnaire);
+		List<String> references = JSONFunctions.getChildReferencesFromQuestionnaire(jsonQuestionnaire);
+		deReference(references, questionnaire);
+		logger.info("Sequences inserted");
```
Comment (nsenave) [inline src/main/java/fr/insee/pogues/persistence/service/QuestionnairesServiceImpl.java:146]: this logging doesn't seem to be in the right place

File: src/main/java/fr/insee/pogues/persistence/service/QuestionnairesService.java [hunk 2]
```diff
@@ -30,7 +30,23 @@ public interface QuestionnairesService {
...
+    JSONObject getQuestionnaireByIDWithReferences(String id) throws Exception;
+
+    /**
+     *
+     * @param jsonQuestionnaire JSON representation of a questionnaire
+     *  @return JSON representation of the questionnaire with references
+     * @throws Exception
+     */
+    JSONObject getQuestionnaireWithReferences(JSONObject jsonQuestionnaire) throws Exception;
+
```
Comment (nsenave) [inline src/main/java/fr/insee/pogues/persistence/service/QuestionnairesService.java:49]: a bit more explanation about what "with references" means would be helpful here

File: src/main/java/fr/insee/pogues/webservice/rest/PoguesPersistence.java [hunk 3]
```diff
@@ -85,9 +87,12 @@ public class PoguesPersistence {
             @ApiResponse(responseCode = "404", description = "Not found")
     })	
 	public ResponseEntity<Object> getQuestionnaire(
-			@PathVariable(value = "id") String id
+			@PathVariable(value = "id") String id,
+			@RequestParam(name = "references", defaultValue = "false") Boolean ref
 	) throws Exception {
```
Comment (nsenave) [inline src/main/java/fr/insee/pogues/webservice/rest/PoguesPersistence.java:92]: java doc about parameters would be helpful

File: src/test/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImplTest.java [hunk 4]
```diff
@@ -83,7 +84,7 @@ void dereference_nullReferences() throws Exception {
 
         //
         PoguesJSONToPoguesJSONDerefImpl deref = new PoguesJSONToPoguesJSONDerefImpl(questionnairesService);
-        assertThrows(NullReferenceException.class, () -> deref.transformAsQuestionnaire(testedInput));
+        assertThrows(PoguesDeserializationException.class, () -> deref.transformAsQuestionnaire(testedInput));
```
Comment (nsenave) [inline src/test/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImplTest.java:87]: The previous error type seems to indicate more clearly what happens in this case: "deserialization exception" doesn't help to understand that the problem is a null reference

**Review by nsenave — COMMENTED** [review]
I like the idea of having the "complete" questionnaire with `GET` `/questionnaire/{id}?references=true` with a default false value. Definitively something that was missing in the API (it was actually possible but in a way that felt twisted)

"references" is a bit confusing to me, yet I don't have a way better naming idea 😄 
A bit more doc around it would be sufficient

**Comment by laurentC35** [comment]
Thanks, I will add docs and we can merged this

File: src/test/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImplTest.java (hunk 4 above)
Comment (laurentC35) [reply src/test/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImplTest.java:87]: Error with the mock, i fix it

**Comment by laurentC35** [comment]
Docs has been added and exceptions corrected.

**Review by nsenave — APPROVED** [review]

**Pull request merged**